### PR TITLE
Connect command

### PR DESCRIPTION
**Pull request by thegrumpylion**
* FileNode to FileNode
* FileNode to PackageNode

### CODE REVIEW

File: pkg/cli/connect.go [hunk 1]
```diff
@@ -14,21 +14,28 @@ var validFileToFileEdges = []string{
...
-	fileToFileEdge string
+	fileToFileEdge    string
+	fileToPackageEdge string
 }{}
 
 func init() {
 	rootCmd.AddCommand(connectCmd)
 	connectCmd.Flags().StringVar(&connectCmdFlags.fileToFileEdge, "fileToFileEdge",
 		"derivedFrom", fmt.Sprintf("Edge to use when connecting FileNode to FileNode. One of %v", validFileToFileEdges))
+	connectCmd.Flags().StringVar(&connectCmdFlags.fileToPackageEdge, "fileToPackageEdge",
```
Comment (GiasemiSh) [inline pkg/cli/connect.go]: Why don't you just use one flag for the edges "--edge" ?

Comment (thegrumpylion) [reply pkg/cli/connect.go]: That is a nice idea

**Comment by thegrumpylion** [comment]
retest this please

File: pkg/cli/connect.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+var connectCmd = &cobra.Command{
+	Use:   "connect",
+	Short: "Connect nodes with specific edges",
+	Long: `Usage: qmstrctl connect <that> <this>...
+Connect to Node <that> Node(s) <this>. In case of multiple edges for the specified types you can use --<type>To<type>Edge flag to specify the edge you want.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		setUpControlService()
+		setUpBuildService()
+		if err := connectCmdRun(cmd, args); err != nil {
```
Comment (GiasemiSh) [inline pkg/cli/connect.go]: You are not using the cmd in connectCmdRun(cmd, args). Better omit that.

File: pkg/cli/connect.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+func getUniqueFileNode(queryNode *service.FileNode) (*service.FileNode, error) {
+	stream, err := controlServiceClient.GetFileNode(context.Background(), queryNode)
+	node, err := stream.Recv()
+	if err != nil {
+		return nil, fmt.Errorf("recv FileNode fail for %v: %v", queryNode, err)
+	}
+	_, err = stream.Recv()
+	if err == nil {
+		return nil, fmt.Errorf("more than one FileNode match %v", queryNode)
```
Comment (GiasemiSh) [inline pkg/cli/connect.go]: Here I would also prompt the user to use a better identifier for the node

File: pkg/cli/connect.go (hunk 2 above)
Comment (thegrumpylion) [reply pkg/cli/connect.go]: The function is consistent with cobra's signature. It doesn't matter if i am using it now or not.

File: pkg/cli/connect.go (hunk 3 above)
Comment (thegrumpylion) [reply pkg/cli/connect.go]: How would you do that?

Comment (GiasemiSh) [reply pkg/cli/connect.go]: Include it in the error message

**Pull request merged**